grammar.js

```diff
@@ -640,6 +646,27 @@ module.exports = grammar({
 
     number: $ => /[\d\.]+/,
 
+    boolean_literal: $ => choice('true', 'false'),
+
+   character_literal: $ => token(seq(
+      '\'',
+      optional(choice(
+        seq('\\', choice(
+          /[^xu]/,
+          /u[0-9a-fA-F]{4}/,
+          /u{[0-9a-fA-F]+}/,
+          /x[0-9a-fA-F]{2}/
+        )),
+        /[^\\'\n]/
+      )),
+      '\''
+    )),
```

Comment:
I went back and forth on this a bunch trying to exactly get what is described [here](https://www.scala-lang.org/files/archive/spec/2.13/01-lexical-syntax.html#character-literals), and finally I just went with a modified version of what tree-sitter-rust was using.